tortoisebeacon/tortoise_beacon.go

```diff
@@ -421,7 +422,49 @@ func (tb *TortoiseBeacon) handleEpoch(ctx context.Context, epoch types.EpochID)
 		log.Uint64("epoch_id", uint64(epoch)))
 }
 
-func (tb *TortoiseBeacon) runProposalPhase(ctx context.Context, epoch types.EpochID) error {
+func (tb *TortoiseBeacon) readProposalMessagesLoop(ctx context.Context, ch chan extendedProposalMessage) {
+	for {
+		select {
+		case <-tb.CloseChannel():
+			close(ch)
+
+			return
+		case em := <-ch:
+			if err := tb.handleProposalMessage(em.message, em.receivedTime); err != nil {
+				tb.Log.With().Error("Failed to handle proposal message",
+					log.String("sender", em.gossip.Sender().String()),
+					log.String("message", em.message.String()),
+					log.Err(err))
+
+				return
+			}
+
+			em.gossip.ReportValidation(ctx, TBProposalProtocol)
+		}
+	}
+}
+
+func (tb *TortoiseBeacon) closePrevChan(epoch types.EpochID) {
```

Comment:
why is this called close prev channel?  it closes proposal channel for epoch `epoch`